lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
```

Comment:
By work I meant be spec compliant. Spec stays to convert this with ToObject, which creates new Number object in case argument is a number, so I think we should not throw. It also mentions that "indexOf function is intentionally generic; it does not require that its this value be an Array object".

http://www.ecma-international.org/ecma-262/7.0/index.html#sec-array.prototype.indexof

http://www.ecma-international.org/ecma-262/7.0/index.html#sec-toobject

However, our current implementation as well as Firefox and Chrome all throw type errors, so it's possible I'm misunderstanding something. Maybe @bterlson can help clarify what the correct behavior should be?